### ISSUE TITLE
log4j, log4j2, logback plugin requirement

### ISSUE THREAD

**Issue by wu-sheng** [wu-sheng/sky-walking#59]
- [x] Use skywalking-api to define plugins for **log4j**, **log4j2**, **logback**
- [x] Plugin will not create a span or log any data ( What is a span, log? please check on [OpenTracing Spec](https://wu-sheng.gitbooks.io/opentracing-io/content/) )
- [x] Major feature is add **traceid** to logs, no matter any log level, patter, etc.

### For example
* origin log
```
[DEBUG] Returning handler method [public org.springframework.web.servlet.ModelAndView com.ai.cloud.skywalking.example.controller.OrderSaveController.save(javax.servlet.http.HttpServletRequest)] 2016-01-19 16:36:53.288 org.springframework.beans.factory.support.DefaultListableBeanFactory 
```

* plugin in active
```
TID:1.0a2.1453192613272.2e0c63e.11144.58.1  [DEBUG] Returning handler method [public org.springframework.web.servlet.ModelAndView com.ai.cloud.skywalking.example.controller.OrderSaveController.save(javax.servlet.http.HttpServletRequest)] 2016-01-19 16:36:53.288 org.springframework.beans.factory.support.DefaultListableBeanFactory 
```

** focus on the prefix of the line **

@zhxiaogg

**Comment by wu-sheng** [comment]
@ascrutae , codes have been pulled. please test to make sure it works.

**Comment by wu-sheng** [comment]
Ref a issue: https://github.com/wu-sheng/sky-walking/issues/71, for log4j2. Hope to be resolved in 2.1-2017

**Comment by wu-sheng** [comment]
All resolved in https://github.com/wu-sheng/sky-walking/issues/62
